### PR TITLE
UI: Replace the icon prop in the Manager API

### LINKED ISSUE #26436: [Bug]: Storybook 8 notifications uses deprecated icons
### Describe the bug

As can be seen in the image:

image

Location: https://github.com/storybookjs/storybook/blob/7a10141273f5b4a31321872d6829a596ee100cb9/code/ui/manager/src/components/notifications/NotificationItem.tsx#L85

### To Reproduce

_No response_

### System

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdedreuille**
Closes https://github.com/storybookjs/storybook/issues/26436

## What I did

In this PR I'm updating the manager API to use a `ReactNode` instead of an object with `name` and `color` for the notification component. This removes the use of the deprecated icons.

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/ui/manager/src/components/notifications/NotificationItem.tsx [hunk 1]
```diff
@@ -76,20 +77,9 @@ const ItemContent: FC<Pick<State['notifications'][0], 'icon' | 'content'>> = ({
...
-      {!icon || (
-        <NotificationIconWrapper>
-          <Icons
-            icon={icon.name as IconsProps['icon']}
-            width={16}
-            height={16}
-            color={icon.color || defaultColor}
-          />
-        </NotificationIconWrapper>
-      )}
```
Comment (ndelangen) [inline code/ui/manager/src/components/notifications/NotificationItem.tsx:92]: This is a breaking change.

I think it's also inconsistent with the types definition for `addNotification`:
https://github.com/storybookjs/storybook/blob/eb6d200a6a0fa9fc4775b267f433704e1c578078/code/lib/types/src/modules/api.ts#L123-L135

File: code/lib/types/src/modules/api.ts [hunk 2]
```diff
@@ -127,10 +127,7 @@ export interface API_Notification {
     headline: string;
     subHeadline?: string | any;
   };
-  icon?: {
-    name: string;
-    color?: string;
-  };
```
Comment (ndelangen) [inline code/lib/types/src/modules/api.ts:133]: This is a breaking change.

File: code/ui/manager/src/components/notifications/NotificationItem.tsx (hunk 1 above)
Comment (cdedreuille) [reply code/ui/manager/src/components/notifications/NotificationItem.tsx:92]: I updated the version to make it backward compatible. I'm checking if this is a valid react component. If it's not, then I'm not showing it.

File: code/lib/types/src/modules/api.ts (hunk 2 above)
Comment (cdedreuille) [reply code/lib/types/src/modules/api.ts:133]: I added both types to make it backward compatible,

**Review by kasperpeulen — APPROVED** [review]
LGTM!

**Pull request merged**